dvc/command/stage.py

```diff
@@ -1,11 +1,109 @@
 import argparse
 import logging
+from itertools import chain, filterfalse
+from typing import TYPE_CHECKING, Dict, Iterable, List
 
 from dvc.command import completion
 from dvc.command.base import CmdBase, append_doc_link, fix_subparsers
 
+if TYPE_CHECKING:
+    from dvc.output.base import BaseOutput
+    from dvc.stage import Stage
+
 logger = logging.getLogger(__name__)
 
+MAX_TEXT_LENGTH = 80
+ELLIPSIS = "…"
+
+
+def generate_description(stage: "Stage") -> str:
+    def part_desc(outs: Iterable["BaseOutput"]) -> str:
+        return ", ".join(out.def_path for out in outs)
+
+    if not stage.deps and not stage.outs:
+        return "No outputs or dependencies."
+
+    if not stage.outs and stage.deps:
+        return "Stage depends on " + part_desc(stage.deps)
+
+    def is_plot_or_metric(out: "BaseOutput"):
+        return bool(out.plot) or bool(out.metric)
+
+    desc: List[str] = []
+
+    outs = list(filterfalse(is_plot_or_metric, stage.outs))
+    if outs:
+        desc.append("Produces " + part_desc(outs))
+
+    plots_and_metrics = list(filter(is_plot_or_metric, stage.outs))
+    if plots_and_metrics:
+        desc.append("Generates " + part_desc(plots_and_metrics))
```

Comment:
"Outputs metric: "

Reply:
That's too long. I'd onl like to add a single verb here.

Reply:
OK. How is it indicated that this refers to metrics or plots? Would be great to have an updated example or summary as suggested by Ivan 🙂

Reply:
Yep I tried this in a test repo I have and it's not clear what the difference between "Produces" (soon Outputs?) and "Generates" is.

```console
$ dvc stage list
ab   Produces ab
abm  Generates ab.met
```

Reply:
Maybe also use "Outputs" and append "metric" or "plot" at the end of the line?

Reply:
Adding metric and plot is one more word for each of the outputs. Would be better to find a single word to explain all these kinds of output.

Reply:
Remember that user will have the output path that is already quite lengthy. I am trying to keep it within a single line.

Reply:
Oops sorry super long reply. I know this is merged so OK. But for the record, keeping "produces" and "generates" seems to me  like the worst alternative. It's unclear why the former refers to regular outputs and the latter to dependencies.

Reply:
I mean metrics/plots

Reply:
\* super late reply 😅